rpc/link.c

```diff
@@ -95,9 +95,29 @@ static const struct m0_fom_type_ops rpc_link_disc_fom_type_ops = {
 	.fto_create = NULL,
 };
 
+/**
+ * Locality function for rpc link connect and disconnect foms.
+ */
 static size_t rpc_link_fom_locality(const struct m0_fom *fom)
 {
-	return 1;
+	/*
+	 * This fom must be scheduled on a different locality than the locality
+	 * to which confc_ready_async_ast() ast is posted by
+	 * m0_confc_ready_cb(), because that ast waits the establishment of the
+	 * for rpc connection initiated by connect_to_confd().
```

Comment:
What if there will be some another subsystem which will schedule on the same locality waiting for the rpc connection - we will have the same issue again?